src/execute.rs

```diff
@@ -111,12 +116,11 @@ impl ProcessReader {
         self.state.lock().unwrap().status = status;
     }
 
-    fn process_output(state: Arc<Mutex<ExecutionState>>, stdout: BufReader<ChildStdout>) -> io::Result<()> {
-        for line in stdout.lines() {
-            let line = line?;
-            // TODO: consider not locking per line...
-            state.lock().unwrap().output.push(line);
-        }
+    fn process_output(state: Arc<Mutex<ExecutionState>>, mut reader: os_pipe::PipeReader) -> io::Result<()> {
+        let mut output = String::new();
+        reader.read_to_string(&mut output)?;
```

Comment:
Why can't this use a `BufReader` and pull line by line like the old code? Is there anything preventing this?

Reply:
Ah yes you're right - I've now amended this to read line by line, and updated the recording. Seems to be working as expected now.